src/main.ts

```diff
@@ -0,0 +1,238 @@
+import * as fs from 'fs';
+import * as core from '@actions/core';
+import * as github from '@actions/github';
+import * as exec from '@actions/exec';
+import * as cache from '@actions/cache';
+import got from 'got';
+
+export enum Inputs {
+  CurrentStatus = 'current-status',
+  SlackWebhook = 'slack-webhook',
+  GithubToken = 'github-token',
+}
+
+const { context } = github;
+const repository = context.repo.repo;
+const cachePrimaryKey = `last-run-status-${context.runId}-${Math.random()
+  .toString(36)
+  .substr(2, 12)}`;
+const cacheRestoreKeys = [`last-run-status-${context.runId}-`];
+const cachePaths = ['last-run-status'];
+
+/**
+ * Gets the last workflow run status
+ *
+ * @returns last run status
+ */
+async function getLastRunStatus() {
+  let lastStatus = '';
+
+  const cacheKey = await cache.restoreCache(
+    cachePaths,
+    cachePrimaryKey,
+    cacheRestoreKeys,
+  );
+
+  if (!cacheKey || (cacheKey && !fs.existsSync(cachePaths[0]))) {
+    core.info('Cache not found, retrieve status from previous run.');
+
+    let headRef;
+
+    if (context.payload.pull_request !== undefined) {
+      headRef = JSON.parse(JSON.stringify(context.payload.pull_request)).head
+        .ref;
+    } else {
+      headRef = context.ref.split('/').pop();
+    }
+
+    core.info(`Branch name: ${headRef}`);
+
+    const githubToken = core.getInput(Inputs.GithubToken);
+    core.exportVariable('GITHUB_TOKEN', `${githubToken}`);
+
+    const options: any = {};
+    options.listeners = {
+      stdout: (data: Buffer) => {
+        lastStatus += data.toString();
+      },
+    };
+
+    await exec.exec(
```

Comment:
I began with this when working with a composite custom action because it was the best solution to manage this. And finally when we moved to a NodeJS custom action, I found it easier to retrieve the needed information about the last completed workflow for the current branch.
Also gh CLI is available by default in our workflow environment.

Whatever it's not a requirement to use `gh` CLI and I would totally understand that we prefer to use API calls only and no execs, so let me know if you want me to adapt this part.

Reply:
I think it's okay to leave as you have it. I ask because I'd imagine it'd be more readable getting back a JSON resopnse as opposed to `grep`ing the stream but what you have is fine. Using `gh` might actually make the developer experience better while testing locally. Not sure what it would be with `got` but when I tested with `gh`, it prompted me to authenticate so the `GITHUB_TOKEN` was automatically available for me which was nice.